Documentation/ceph-upgrade.md

```diff
@@ -234,6 +234,12 @@ A new `CephClient` CRD is included in v1.2 and the CSI driver privileges changed
 kubectl apply -f upgrade-from-v1.1-apply.yaml
 ```
 
+Update the ClusterRole required for attacher and resizer of CSI driver.
+
+```sh
+kubectl apply -f upgrade-from-v1.2-apply.yaml
```

Comment:
This section is only for upgrades from v1.1. There is a section above for patch upgrade releases. So I have some concerns with this CSI 2.0 upgrade. The patch releases have traditionally not required RBAC updates.

If we go ahead with this, we would need to add this above in the patch release section for those upgrading from v1.2 releases before v1.2.3. But I'm thinking the following instead...

The CSI v2.0 is a big update to include in a patch release. Besides requiring RBAC it will be a risky change for someone expecting a less risky v1.2.x patch release. How about the following?
- In Rook master, CSI v2.0 would be the default as you have in this PR. 
- In the release-1.2 branch:
   - For upgraded deployments keep CSI v1.2 as the default (the default in the code would still be v1.2). If someone wants to run the 2.0 operator we would have additional steps in the upgrade guide in order to start using it by updating your operator deployment and applying the RBAC.
   - For new deployments CSI v2.0 would be the default (the default would already be set in operator.yaml and the RBAC would already be included in common.yaml)

Reply:
> This section is only for upgrades from v1.1. There is a section above for patch upgrade releases. So I have some concerns with this CSI 2.0 upgrade. The patch releases have traditionally not required RBAC updates.
> 
> If we go ahead with this, we would need to add this above in the patch release section for those upgrading from v1.2 releases before v1.2.3. But I'm thinking the following instead...
> 
> The CSI v2.0 is a big update to include in a patch release. Besides requiring RBAC it will be a risky change for someone expecting a less risky v1.2.x patch release. How about the following?
> 
> * In Rook master, CSI v2.0 would be the default as you have in this PR.
> * In the release-1.2 branch:
>   
>   * For upgraded deployments keep CSI v1.2 as the default (the default in the code would still be v1.2). If someone wants to run the 2.0 operator we would have additional steps in the upgrade guide in order to start using it by updating your operator deployment and applying the RBAC.
>   * ~For new deployments CSI v2.0 would be the default (the default would already be set in operator.yaml and the RBAC would already be included in common.yaml)~
> 
> To revise my last statement, I'd recommend CSI v1.2 remain the default for upgraded **and** new Rook v1.2 clusters. If we were to override the default in operator.yaml as I suggested, we would be in a situation where everyone would be overriding the default. Instead, overriding the default should be an explicit action that the user makes intentionally. So we just need to document how Rook v1.2 users can enable the CSI v2.0 driver.


The RBAC changes are not exactly for the cephcsi v2.0.0. This is for the kubernetes sidecar containers. if we don't want the RBAC changes basically two things we have to do 

* Don't use node-driver-registrar v2.1.0 https://github.com/kubernetes-csi/external-attacher/blob/master/CHANGELOG-2.0.md#changelog-since-v120 because it is not compatible with 1.x driver ( i have to add if check  for some flags in attacher container if it is v1.2.x)

* Resizer won't work by default  if RBAC is not updated